### ISSUE TITLE
[FEA]: Integrate CUDASTF with CudaX

### ISSUE THREAD

**Issue by alliepiper** [NVIDIA/cccl#2533]
Setup CI, tests, examples, header tests, etc.

```[tasklist]
### Tasks
- [x] Copy headers to cudax
- [x] Rename headers from `.h` to `.cuh`
- [x] Limit scope of `--extended-lambda` flag to only stf-related compilation units in cudax (currently global to unblock port)
- [x] Update include paths to use global includes, .cuh extensions
- [x] Add tests
- [x] Add examples
- [x] Clang-format
- [x] Add CUB (and other CCCL?) deps to tests/examples as needed
- [x] unittested headers
- [x] tests/static_error_checks
- [x] Integrate documentation
- [x] check-linking
- [ ] https://github.com/NVIDIA/cccl/issues/2652
- [ ] Replace CUDASTF_HOST_DEVICE, etc with _CCCL equivalents
- [ ] Remove `--expt-relaxed-constexpr` flag from cudax build (temporarily added to unblock port)
- [ ] Replace `supplemental_std_experimental` utilities with libcudacxx where possible.
- [ ] MSVC support
```

**Comment by caugonnet** [comment]
In the TODO, we also need (as a low priority) to connect the doxygen we have to the generated doc for cudax (sphinx doc ?)
